lib/resourcemerge/batch_test.go

```diff
@@ -0,0 +1,103 @@
+package resourcemerge
+
+import (
+	"testing"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+func TestEnsureJob(t *testing.T) {
+	labelSelector := metav1.LabelSelector{}
+	tests := []struct {
+		name     string
+		existing batchv1.Job
+		required batchv1.Job
+
+		expectedModified bool
+		expected         batchv1.Job
+	}{
+		{
+			name: "different backofflimit count",
+			existing: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+			required: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(3)}},
+
+			expectedModified: true,
+			expected: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(3)}},
+		},
+		{
+			name: "same backofflimit count",
+			existing: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+			required: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+
+			expectedModified: false,
+			expected: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+		},
+		{
+			name: "required-Selector not nil",
+			existing: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Selector:       &labelSelector,
+					ManualSelector: boolPtr(false)}},
+			required: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Selector:       &labelSelector,
+					ManualSelector: boolPtr(true)}},
+
+			expectedModified: false,
+			expected: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Selector:       &labelSelector,
+					ManualSelector: boolPtr(false)}},
+		},
+	}
+
+	for i, test := range tests {
+		if i == len(tests)-1 {
```

Comment:
This is a brittle switch.  Can you add `expectedPanic bool` or some such to your test-case structure and switch on that instead?

Reply:
Agree, not good!

Reply:
6b58e7a has `expectedPanic`.